### DISCUSSION TITLE
Help with json

### DISCUSSION THREAD

**Discussion by ghost** [TomWright/dasel#105] [Q&A] [answered]
Hello @TomWright,

I hope you can help me. Can I create the below json with Dasel?

```
{
  "Comment": "CREATE/DELETE/UPSERT a record",
    "Changes": [{
    "Action": "UPSERT",
      "ResourceRecordSet": {
        "Name": "metrics.try.my.domain",
        "Type": "SRV",
        "TTL": 300,
        "ResourceRecords": [
          { "Value": "1 10 30901 metrics-peers.try.my.domain"}
        ]
      }
    }]
}
```
Thanks!

**Comment by TomWright** [comment, marked as answer]
Hey @ws-way,

There are a few ways you could put this together - dasel lets you put strings, ints and bools one at a time or you can put entire documents.

The following will create the document for you. This is possible because dasel interprets the selectors and will create empty objects/arrays for you where possible.

```
echo '' | \
dasel put string -p json '.Comment' 'CREATE/DELETE/UPSERT a record' | \
dasel put string -p json '.Changes.[0].Action' 'UPSERT' | \
dasel put string -p json '.Changes.[0].ResourceRecordSet.Name' 'metrics.try.my.domain' | \
dasel put string -p json '.Changes.[0].ResourceRecordSet.Type' 'SRV' | \
dasel put int -p json '.Changes.[0].ResourceRecordSet.TTL' '300' | \
dasel put string -p json '.Changes.[0].ResourceRecordSet.ResourceRecords.[0].Value' '1 10 30901 metrics-peers.try.my.domain'
```

```json
{
  "Changes": [
    {
      "Action": "UPSERT",
      "ResourceRecordSet": {
        "Name": "metrics.try.my.domain",
        "ResourceRecords": [
          {
            "Value": "1 10 30901 metrics-peers.try.my.domain"
          }
        ],
        "TTL": 300,
        "Type": "SRV"
      }
    }
  ],
  "Comment": "CREATE/DELETE/UPSERT a record"
}
```

On a side note, you may need to be on at least `v1.13.5` for this to work as a bug was recently fixed.

**Reply by link4531** [reply]
this example works fine when using -p json or -p yaml
however doesn't work when using -p xml?

dasel returns error: could not load input: could not unmarshal data: EOF. 

however if I write out all the intended values first in yaml/json then convert it to xml after I get the intended structure.

**Reply by TomWright** [reply]
Hey @link4531,

Thanks for reporting that. I've created an issue here: https://github.com/TomWright/dasel/issues/131

**Comment by ghost** [comment]
Thanks @TomWright!
